storage/src/index/storage.rs

```diff
@@ -18,9 +24,22 @@ pub struct ValueIterator<'a, V> {
     next: Option<&'a Record<V>>,
 }
 
-// An iterator over all values associated with a translated key, allowing for mutation.
-pub struct MutableValueIterator<'a, V> {
-    next: Option<&'a mut Record<V>>,
+/// An iterator over all values associated with a translated key, allowing for mutation and deletion
+/// of the current element.
+pub struct DeleteValueIterator<'a, K, V> {
```

Comment:
nit: I prefer to put all `DeleteValueIterator` logic together (rather than defining structs/then funcs). Not sure how you feel.

Reply:
agree, done